spdb/spatialdb/object_indices.py

```diff
@@ -1072,6 +1072,45 @@ def update_last_partition_key(self, key, chunk_num, version=0):
             raise SpdbError("Failed to update {} for ID: {}".format(LAST_PARTITION_KEY, key),
                 ErrorCodes.OBJECT_STORE_ERROR)
 
+    def get_cuboids_from_id(self, resource, resolution, id, version=0):
+        """
+        Get a list of cuboid bounds that contains the object labeled with id.
+
+        Args:
+            resource (project.BossResource): Data model info based on the request or target resource
+            resolution (int): the resolution level
+            id (uint64|string): object's id
+            version (optional[int]): Version - reserved for future use.
+        Returns:
+           (dict|None) : Extents of cuboids in XYZ containing ID or None if the id is not found.
+
+        Raises:
+            (SpdbError): Can't talk to id index database or database corrupt.
+        """
+
+        cuboid_list = []
+
+        [x_cube_dim, y_cube_dim, z_cube_dim] = CUBOIDSIZE[resolution]
+        obj_keys = self.get_cuboids(resource, resolution, id)
+
+        if len(obj_keys) == 0:
+            return None
+
+        for key in obj_keys:
+            morton = int(key.split('&')[6])
```

Comment:
Recommend using `get_object_key_parts()` here just in case we make changes to the key format.